Task to check base C# knowledge completed

fjord-code opened the pull request:
# Completed the task to check base C# knowledge.
- Created structures to store card data.
- Implemented logic to input card data from the console.
- Implemented game logic.

fjord-code committed:
Logic

Added structures to build the logic on.

fjord-code committed:
Implemented logic

Finished logic implementation.

AnnaGolosova commented on Program.cs:
```diff
@@ -1,10 +1,185 @@
+
+    struct Color
+    {
+        private static Dictionary<Colors, int> _colorsSortOrder = new Dictionary<Colors, int> 
```
Here and below summary comments are missing

AnnaGolosova commented on Program.cs:
```diff
@@ -1,10 +1,185 @@
+            { Colors.Purple, 100 },
+        };
+
+        public char Name { get; private set; }
```
Why Name stored as char? Why not as `Colors` enum?

AnnaGolosova commented on Program.cs:
```diff
@@ -1,10 +1,185 @@
+
+        private static int GetColorsValue(Colors color)
+        {
+            if (!_colorsSortOrder.ContainsKey(color))
```
You can use `TryGetValue` method, it will be more clear and performant

AnnaGolosova commented on Program.cs:
```diff
@@ -1,10 +1,185 @@
+
+        public void Add(Card card)
+        {
+            _cards.Add(card);
```
Is there a limit for cards number in `CardCombination`?

AnnaGolosova commented on Program.cs:
```diff
@@ -1,10 +1,185 @@
+
+            var rawInput = Console.ReadLine();
+            var rawValues = rawInput.Split(' ');
+            var value = int.Parse(rawValues[0]);
```
What if `Parse` method ended with error? I cannot see errors handling

AnnaGolosova commented on Program.cs:
```diff
@@ -1,10 +1,185 @@
+            );
+        }
+
+        static void AddCardToCombination(ref CardCombination cardCombination, Card card)
```
Why `cardCombination` is marked as `ref` here?

fjord-code committed:
Refactored code

- Added summary comments.
- Added Equals and GetHashCode methods implementation for the structures.
- Refactored code.

fjord-code committed:
Refactored code

- Refactored code.
- Added new constructor to the Card structure.
- Started moving game logic to MVVM.

fjord-code committed:
Implemented MVVM approach

- Moved entities to related cs files.
- Created interfaces for a game and a presenter.
- Implemented a controller.
- Refactored code.

fjord-code committed:
Implemented MVVM approach

- Changed the IPresenter interface.
- Created the Presenter class that implemets the IPresenter interface.

fjord-code committed:
Implemented MVVM

- Added the game logic.
- Refactored code.
